File: backend-java/src/main/java/com/ilena/backend/model/Treino.java
package com.ilena.backend.model;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Treino {
    private Integer id;

    @NotBlank(message = "Nome do treino é obrigatório")
    private String nome;

    @NotBlank(message = "Tipo do treino é obrigatório")
    private String tipo;

    @Positive(message = "Duração deve ser maior que 0")
    private Integer duracaoMinutos;

    private String observacoes;
}

File: app/src/main/java/com/ilena/app/model/Treino.kt
package com.ilena.app.model

import com.google.gson.annotations.SerializedName

data class TreinoRequest(
    val nome: String,
    val tipo: String,
    @SerializedName("duracao_minutos") val duracaoMinutos: Int,
    val observacoes: String? = null,
)

data class TreinoResponse(
    val id: Int,
    val nome: String,
    val tipo: String,
    @SerializedName("duracao_minutos") val duracaoMinutos: Int,
    val observacoes: String?,
    val mensagem: String,
)
